components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -218,6 +218,31 @@ export const rewardsPanelReducer = (state: RewardsExtension.State | undefined, a
         chrome.braveRewards.getGrant()
         break
       }
+    case types.SAVE_RECURRING_DONATION:
+      {
+        let newAmount = payload.newAmount
+        let publisherKey = payload.publisherKey
+        if (publisherKey == null) {
+          break
+        }
+        chrome.braveRewards.saveRecurringDonation(publisherKey, newAmount)
```

Comment:
let's check if amount is number and positive as well

Reply:
done